src/main/java/org/folio/innreach/batch/contribution/InitialContributionJobConsumerContainer.java

```diff
@@ -0,0 +1,114 @@
+package org.folio.innreach.batch.contribution;
+
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.folio.innreach.batch.contribution.listener.ContributionExceptionListener;
+import org.folio.innreach.domain.dto.folio.inventorystorage.InstanceIterationEvent;
+import org.folio.innreach.external.exception.ServiceSuspendedException;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.listener.ContainerProperties;
+import org.springframework.kafka.listener.DefaultErrorHandler;
+import org.springframework.util.backoff.BackOff;
+import org.springframework.util.backoff.FixedBackOff;
+
+import java.net.SocketTimeoutException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.kafka.clients.consumer.ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG;
+
+@Log4j2
+@Setter
+@RequiredArgsConstructor
+public class InitialContributionJobConsumerContainer {
+
+  public static Map<String, ConcurrentMessageListenerContainer<String, InstanceIterationEvent>> consumersMap =
+    new HashMap<>();
+
+
+  private final Map<String,Object> consumerProperties;
+  private final String topic;
+  private final Deserializer<String> keyDeserializer;
+  private final Deserializer<InstanceIterationEvent> valueDeserializer;
+
+  private final Long interval;
+
+  private final Long maxAttempts;
+
+  private static final int CONCURRENT_CONTAINER_SIZE = 2;
```

Comment:
Maybe just `CONCURENCY` would be better. Not sure if "size" really makes much sense here.